src/tile_hires_stencil.cc

```diff
@@ -0,0 +1,419 @@
+            printf("centerTileGlobalX=%i, centerTileGlobalY=%i\n", centerTileGlobalX, centerTileGlobalY);
+            printf("kekekek border=%i\n", gTileBorderInitialized);
+            exit(102);
+        }
+        if (
+            centerTileGlobalX + squares_screen_width_half * square_width >= square_width * square_grid_width
+            || centerTileGlobalY + squares_screen_height_half * square_height >= square_height * square_grid_height) {
+            printf("centerTileGlobalX=%i, centerTileGlobalY=%i\n", centerTileGlobalX, centerTileGlobalY);
+            exit(103);
+        }
+    };
+#endif
+
+    int squareX = centerTileGlobalX / square_width;
+    int squareY = centerTileGlobalY / square_height;
+
+    int horizonal_start_full = squareX - squares_screen_width_half + 1;
+    int horizonal_end_full = squareX + squares_screen_width_half;
+
+    int horizontal_start = part != MarkOnlyPart::RIGHT
+        ? horizonal_start_full
+        : horizonal_end_full - squares_per_horizontal_move;
+    int horizontal_end = part != MarkOnlyPart::LEFT
+        ? horizonal_end_full
+        : horizonal_start_full + squares_per_horizontal_move;
+
+    int vertical_start_full = squareY - squares_screen_height_half;
+    int vertical_end_full = squareY + squares_screen_height_half;
+
+    int vertical_start = part != MarkOnlyPart::DOWN
+        ? vertical_start_full
+        : vertical_end_full - squares_per_vertical_move;
+    int vertical_end = part != MarkOnlyPart::UP
+        ? vertical_end_full
+        : vertical_start_full + squares_per_vertical_move;
+
+    for (int x = horizontal_start; x <= horizontal_end; x++) {
+        for (int y = vertical_start; y <= vertical_end; y++) {
+            visible_squares[gElevation][x][y] = true;
+        }
+    }
+}
+
+void tile_hires_stencil_on_center_tile_or_elevation_change()
+{
+    if (!gIsTileHiresStencilEnabled) {
+        return;
+    }
+
+    if (!gTileBorderInitialized) {
+        return;
+    };
+
+    if (visited_tiles[gElevation][gCenterTile]) {
+        debugPrint("tile_hires_stencil_on_center_tile_or_elevation_change tile was visited gElevation=%i gCenterTile=%i so doing nothing\n",
+            gElevation, gCenterTile);
+
+        return;
+    };
+
+    debugPrint("tile_hires_stencil_on_center_tile_or_elevation_change non-visited tile gElevation=%i gCenterTile=%i\n",
+        gElevation, gCenterTile);
+
+    clean_cache_for_elevation(gElevation);
+
+    struct TileToVisit {
+        int tile;
+        MarkOnlyPart part;
+    };
+
+    std::vector<struct TileToVisit> tiles_to_visit {};
+    tiles_to_visit.reserve(7000);
+
+    tiles_to_visit.push_back({ gCenterTile, MarkOnlyPart::FULL });
+
+    int visited_tiles_count = 0;
+
+    auto screen_diff = get_screen_diff();
+
+    while (!tiles_to_visit.empty()) {
+        auto tileInfo = tiles_to_visit.back();
```

Comment:
Question to C++ gurus: do we copy element (which is struct) here or just getting the reference? If it is a reference then doing `pop_back` should make it destroyed, is not it?